### PR TITLE
Add HRV full disk navigation for `seviri_l1b_native`-reader

### LINKED ISSUE #975: Add HRV navigation to `seviri_l1b_native`-reader

### PR DESCRIPTION
**Pull request by ameraner**
PR implements the navigation for the SEVIRI HRV full disk case in the Native format reader.
The HRV channel covers two separated geographical areas; when ordering the full disk data, the two areas are stacked together and need to be navigated to the original correct geolocations by reading the correspondent metadata and defining two area_definitions. This is now implemented for the Native reader, which now covers all data order combinations.

 - [x] Closes #975 
 - [x] Tests added and test suite added to parent suite 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented
 - [x] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

File: satpy/readers/seviri_l1b_native.py [hunk 1]
```diff
@@ -307,11 +346,47 @@ def get_area_extent(self, dsid):
         # When dealing with HRV channel and full disk, area extent is
         # in two pieces
         if (dsid.name == 'HRV') and self.mda['is_full_disk']:
-            # NOTE: Implement HRV full disk area_extent
-            # NotImplementedError does not catch this, it must
-            # be used elsewhere already
-            msg = 'HRV full disk area extent not implemented.'
-            raise RuntimeError(msg)
+
+            center_point = HRV_NUM_COLUMNS / 2
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: please move `center_point` calculation under the `if`-statement starting on line 332. `column_step`, and `line_step` could be calculated right after the `if`-statement using the `reference_grid` variable. `column_step` starting on line becomes redundant then.

File: satpy/readers/seviri_l1b_native.py [hunk 2]
```diff
@@ -307,11 +346,47 @@ def get_area_extent(self, dsid):
...
+            column_step = data15hd['ImageDescription'][
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: `column_step` and `line_step` is calculated in `if` and `else` branches, can be moved outside. see comment above.

File: satpy/readers/seviri_l1b_native.py [hunk 3]
```diff
@@ -343,9 +418,8 @@ def get_area_extent(self, dsid):
 
         return area_extent
 
-    def get_dataset(self, dsid, info,
-                    xslice=slice(None), yslice=slice(None)):
-
+    def get_dataset(self, dsid, info):
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: For clarity, please rename `dsid` to `dataset_id` and `info` to `dataset_info`. Apply the same elsewhere in the reader and tests, if/when necessary.

File: satpy/tests/reader_tests/test_seviri_l1b_native.py [hunk 4]
```diff
@@ -101,16 +93,28 @@
     }
 }
 
-TEST_AREA_EXTENT_EARTHMODEL1_HRV_FULLDISK = None
+TEST_AREA_EXTENT_EARTHMODEL1_HRV_FULLDISK = {
+    'earth_model': 1,
+    'dsid': DatasetID(name='HRV'),
```
Comment (sjoro) [inline satpy/tests/reader_tests/test_seviri_l1b_native.py]: See earlier comment about `dsid` and rename to `dataset_id`

File: satpy/readers/seviri_l1b_native.py [hunk 5]
```diff
@@ -254,18 +259,52 @@ def get_area_def(self, dsid):
             pdict['a_name'] = 'geosmsg_hrv'
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: For clarity, please change `geosmsg_hrv` to `geos_seviri_hrv`

File: satpy/readers/seviri_l1b_native.py [hunk 6]
```diff
@@ -254,18 +259,52 @@ def get_area_def(self, dsid):
...
             pdict['a_desc'] = 'MSG/SEVIRI high resolution channel area'
             pdict['p_id'] = 'msg_hires'
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: For clarity, please change `msg_hires` to 'seviri_hrv'

File: satpy/readers/seviri_l1b_native.py [hunk 7]
```diff
@@ -254,18 +259,52 @@ def get_area_def(self, dsid):
...
+                # order of areas is flipped w.r.t. the hrit reader due to the flipping of the data in get_dataset
+                area = geometry.StackedAreaDefinition(upper_area, lower_area)
+                area = area.squeeze()
+            else:
+                area = get_area_definition(pdict, self.get_area_extent(dsid))
+
         else:
             pdict['nlines'] = self.mda['number_of_lines']
             pdict['ncols'] = self.mda['number_of_columns']
             pdict['a_name'] = 'geosmsg'
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: Same as earlier

File: satpy/readers/seviri_l1b_native.py [hunk 8]
```diff
@@ -254,18 +259,52 @@ def get_area_def(self, dsid):
...
             pdict['a_desc'] = 'MSG/SEVIRI low resolution channel area'
             pdict['p_id'] = 'msg_lowres'
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: Please change `msg_lowres` to `seviri_visir`

**Comment by ameraner** [comment]
Added a new commit responding to Sauli's review.

File: satpy/readers/seviri_l1b_native.py [hunk 9]
```diff
@@ -245,34 +245,34 @@ def _read_trailer(self):
...
         pdict['h'] = self.mda['projection_parameters']['h']
         pdict['ssp_lon'] = self.mda['projection_parameters']['ssp_longitude']
 
-        if dsid.name == 'HRV':
+        if dataset_id.name == 'HRV':
             pdict['nlines'] = self.mda['hrv_number_of_lines']
             pdict['ncols'] = self.mda['hrv_number_of_columns']
-            pdict['a_name'] = 'geosmsg_hrv'
+            pdict['a_name'] = 'geos_seviri_hrv'
             pdict['a_desc'] = 'MSG/SEVIRI high resolution channel area'
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_native.py]: Please make the `a_desc` consistent through out the reader and the tests.

**Pull request merged**